File: bancoDev/src/app/services/auth.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environment/environment';
import { ApiResponse } from '../interfaces/ApiResponse';
import { AuthResponse } from '../interfaces/login/AuthResponse';
import { ClienteCrearRequest } from '../interfaces/cliente/ClienteCrearRequest';
import { ClienteResponse } from '../interfaces/cliente/ClienteResponse';

@Injectable({
  providedIn: 'root'
})

export class AuthService {

  private api:string = `${environment.url}/auth`;

  constructor( private http: HttpClient ) { }

  iniciarSesion(correo:string, password:string): Observable<ApiResponse<AuthResponse>> {
    const params = new HttpParams().set('correo', correo).set('password', password);
    return this.http.get<ApiResponse<AuthResponse>>(`${this.api}/iniciarSesion`, {params})
  }

  crearCuenta(cliente:ClienteCrearRequest): Observable<ApiResponse<ClienteResponse>> {
    return this.http.post<ApiResponse<ClienteResponse>>(`${this.api}/crearCuenta`, cliente)
  }

}

File: BancoDev - BackEnd/8 MicroservicioAuth/src/main/java/com/bancoDev/services/AuthService.java
package com.bancoDev.services;

import com.bancoDev.client.models.ClienteCrearRequest;
import com.bancoDev.client.models.ClienteResponse;
import com.bancoDev.models.ApiResponse;
import com.bancoDev.models.AuthResponse;

public interface AuthService {

    ApiResponse<AuthResponse> iniciarSesion(String correo, String password);
    ApiResponse<ClienteResponse> crearCuenta(ClienteCrearRequest cliente);

}
